ui/admin/app/routes/scopes/scope/roles/role/manage-scopes/index.js

```diff
@@ -18,4 +40,34 @@ export default class ScopesScopeRolesRoleManageScopesIndexRoute extends Route {
       controller.set('showCheckIcon', false);
     }
   }
+
+  // =actions
+
+  /**
+   * Triggers confirm pop-up only when user has made changes and is trying to navigate away from current route.
+   * @param {object} transition
+   */
+  @action
+  async willTransition(transition) {
```

Comment:
I see that you needed these in 3 routes, just out of curiosity, was it not possible to directly modify the models for these role grants (instead of using a `selectedItems`) so that way we can have it just be rolled back to the original array? Was it too complicated?

Reply:
I briefly considered this approach but then decided not to go that route because the grant_scopes_ids field is readOnly and thought it would go against existing patterns to save intermediate state in that field. However, i think it could def. work and produce a cleaner result. Im going to try this approach out.

Reply:
Is `grant_scopes_id` readonly? I'm not sure why that would be the case

Reply:
woopsies i think when i was working on this i looked at the wrong field, i looked again and it is not readOnly.

Reply:
refactored to access the field in the model directly.